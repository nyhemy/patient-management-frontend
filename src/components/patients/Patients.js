import React, { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import loadImg from '../ajax-loader.gif';
import styles from './Patients.module.css';
import { get } from '../Requests';
import Patient from '../patient/Patient';

const Patients = () => {
  const history = useHistory();
  const [loading, setLoading] = useState(false);
  const [errorMsg, setErrorMsg] = useState('');

  const [patients, setPatients] = useState([]);

  useEffect(() => {
    setErrorMsg('');
    setLoading(true);
    get('http://localhost:8080/patients')
      .then((response) => {
        setLoading(false);
        setPatients(response.data);
      })
      // eslint-disable-next-line no-unused-vars
      .catch((error) => {
        setLoading(false);
        setErrorMsg('Oops something went wrong');
      });
  }, []);

  const createPatient = () => {
    history.push('/patients/create');
  };

  return (
    <div className={styles.center}>
      <h2>Patients</h2>
      {errorMsg && <h3 className={styles.error}>{errorMsg}</h3>}
      <div className={styles.row}>
        {loading
          ? <img src={loadImg} alt="loading..." />
          : errorMsg === '' && (
          <>
            <div><button type="button" onClick={createPatient}>Create</button></div>
            {patients.map((data) => (
              <div className={styles.column} key={data.id}>
                <Patient
                  id={data.id}
                  firstName={data.firstName}
                  lastName={data.lastName}
                  age={data.age}
                  gender={data.gender}
                />
              </div>
            ))}
          </>
          )}
      </div>
    </div>
  );
};

export default Patients;
