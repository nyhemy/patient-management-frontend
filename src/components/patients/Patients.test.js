import React from 'react';
import '@testing-library/jest-dom/';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, screen, waitForElement, waitFor, getByText } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router, Route, Switch } from 'react-router-dom';
import fetchMock from 'jest-fetch-mock';
import Patients from './Patients';

fetchMock.enableMocks();

beforeEach(() => {
  fetch.resetMocks();
  const test = JSON.stringify([
    {
      id: 1,
      firstName: 'Jane',
      lastName: 'Doe',
      email: 'jd@gmail.com',
      street: 'Drew Dr',
      city: 'Turboville',
      state: 'MA',
      postal: '01545',
      age: 23,
      height: 64,
      weight: 112,
      insurance: 'Turbosure',
      gender: 'female',
      ssn: '000-00-0000'
    },
    {
      id: 2,
      firstName: 'Bob',
      lastName: 'Dude',
      email: 'bd@gmail.com',
      street: 'Dude Dr',
      city: 'Dudeville',
      state: 'NH',
      postal: '00000',
      age: 25,
      height: 70,
      weight: 144,
      insurance: 'Bubba',
      gender: 'male',
      ssn: '000-00-0001'
    }
  ]);

  console.log(test);
  fetch.mockResponse(test);

  // fetch.mockResponseOnce(JSON.stringify([
  //   {"id":1,
  //   "firstName":"Jane",
  //   "lastName":"Doe",
  //   "email":"jd@gmail.com",
  //   "street":"Drew Dr",
  //   "city":"Turboville",
  //   "state":"MA",
  //   "postal":"01545",
  //   "age":23,
  //   "height":64,
  //   "weight":112,
  //   "insurance":"Turbosure",
  //   "gender":"female",
  //   "ssn":"000-00-0000"},
  //   {"id":2,"firstName":"Jason","lastName":"Bourne","email":"null@gmail.com","street":"REDACTED","city":"REDACTED","state":"AK","postal":"01545","age":40,"height":73,"weight":165,"insurance":"REDACTED","gender":"male","ssn":"000-00-0000"},{"id":3,"firstName":"Zero","lastName":"Encounters","email":"zero@gmail.com","street":"Place St","city":"Ether","state":"FL","postal":"01545","age":25,"height":67,"weight":180,"insurance":"Weewoo Gotchu Inc","gender":"male","ssn":"000-00-0000"}
  // ]));
});

afterEach(cleanup);

const history = createMemoryHistory();

const renderWithRouter = (component, url) => {
  history.push(url);

  return {
    ...render(
      <Router history={history}>
        <Switch>
          <Route
            path={url}
            render={() => component}
            exact
          />
        </Switch>
      </Router>
    )
  };
};

test('renders w/o crashing', async () => {
  const { getByText, getAllByText } = render(
    <Router history={history}>
      <Patients />
    </Router>
  );

  await waitFor(() => getByText('Oops something went wrong'));
  expect(screen.queryByText('Oops something went wrong')).toBeTruthy();
});

it('renders correctly and matches snapshot', () => {
  const { asFragment } = render(<Patients />);

  expect(asFragment(<Patients />)).toMatchSnapshot();
});

it('renders patients data', async () => {
  // fetch.mockResponse(JSON.stringify([
  //   {
  //     id: 1,
  //     firstName: 'Jane',
  //     lastName: 'Doe',
  //     email: 'jd@gmail.com',
  //     street: 'Drew Dr',
  //     city: 'Turboville',
  //     state: 'MA',
  //     postal: '01545',
  //     age: 23,
  //     height: 64,
  //     weight: 112,
  //     insurance: 'Turbosure',
  //     gender: 'female',
  //     ssn: '000-00-0000'
  //   }
  // ]));
  const { getByText, getAllByText } = renderWithRouter(<Patients />, '/patients');

  await waitFor(() => getByText('Name: Jane Doe'));
  // expect(screen.queryByText('name', { exact: false })).toBeTruthy();
  // expect(screen.queryByText('404 Not Found')).toBeTruthy();
  // expect(screen.queryByText('Age: 23')).toBeTruthy();
  // expect(screen.queryByText('Gender: female')).toBeTruthy();
});

//*[@id="root"]/div/div/div[2]/div/div/div[1]