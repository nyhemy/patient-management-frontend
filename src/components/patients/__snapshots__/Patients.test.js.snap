// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="center"
  >
    <h2>
      Patients
    </h2>
    <div
      class="row"
    >
      <img
        alt="loading..."
        src="ajax-loader.gif"
      />
    </div>
  </div>
</DocumentFragment>
`;
